File: VB/FilterButtonvisibilityConverter.vb
Imports System
Imports System.Windows.Data
Imports System.Windows.Markup

Namespace E20031
    Public Class FilterButtonvisibilityConverter
        Inherits MarkupExtension
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
            Dim fieldName As String = value.ToString()
            Return fieldName = "Text"
        End Function
        Public Overrides Function ProvideValue(ByVal serviceProvider As IServiceProvider) As Object
            Return Me
        End Function
        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace

File: CS/FilterButtonvisibilityConverter.cs
using System;
using System.Windows.Data;
using System.Windows.Markup;

namespace E20031 {
    public class FilterButtonvisibilityConverter : MarkupExtension, IValueConverter {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            string fieldName = value.ToString();
            return fieldName == "Text";
        }
        public override object ProvideValue(IServiceProvider serviceProvider) {
            return this;
        }
        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            throw new NotImplementedException();
        }
    }
}